### DISCUSSION TITLE
RemafoX command line tool not installed, download & set up app from https://remafox.app

### DISCUSSION THREAD

**Discussion by TopScrech** [FlineDev/RemafoX#97] [Help / FAQ] [answered]
Xcode Cloud gives this yellow warning:
`RemafoX command line tool not installed, download & set up app from https://remafox.app`
and the enum file doesn't update when I add new strings (witch somehow appear in the Localizable file)

After googling for an hour I haven't found any information about this RemafoX command line tool

**Comment by Jeehut** [comment, marked as answer]
Hi @TopScrech, thank you for asking this question. It seems like you've deployed an app to Xcode Cloud which was set up for the RemafoX Mac app to streamline the developer experience when adding localizable Strings during development. For the developer convenience, the build script was added so localizations are updated automatically on each build during development.

The warning is shown to you because the RemafoX app seems not to be installed & configured on the Xcode Cloud machine the test is running on. The command line tool is shipped with the RemafoX app. But that's not a reason to worry. In fact, the RemafoX command line tool is intended for developer machines only and wouldn't make any sense to run on CI machines, except maybe for the linting feature. But for now, there's no tool intended for usage on CI servers, thus you can simply ignore the warning or adjust the build script within the project to not print the warning if it detects that it's running on your CI.

According to [this document](https://developer.apple.com/documentation/xcode/environment-variable-reference) an environment value named `CI` is set to `TRUE` if you're running build scripts on Xcode Cloud. So you could simply wrap the `echo` in the build script with a check like this to turn the warning off on CI builds:

```
if [ "${CI}" != "TRUE" ]; then
   echo "warning: RemafoX command line tool not installed, download & set up app from https://remafox.app"
fi
```

**Reply by TopScrech** [reply]
Thank you so much for your assistance! I'm still facing the second issue unfortunately. The enum file isn't updating as expected. When I add a new string, it replaces for example "Name" on Loc.Name and adds translations to the Localizable file. However, the enum file remains unchanged, resulting in the error `Type 'Loc' (aka 'Res.Str') has no member 'Name'`. Could you provide guidance on how to resolve it?

**Reply by Jeehut** [reply]
@TopScrech Glad that I could help resolve the first issue.

As for the second, I need more information to help you. Could you please provide more details about how you are adding a new key (manually to the Strings file? using the Add Translation window?), maybe provide a full example of what your keys looks like (e.g. a screenshot from the Add Translation window)? Also, please report your version of RemafoX, macOS and Xcode and check the path of the resource file you've set up in the config file. It would really be best if you reported this issue as a [bug report](https://github.com/FlineDev/RemafoX/issues/new?assignees=&labels=Bug+Report&template=bug_report.md&title=%5BBug%5D+%3Cyour+bug+title%3E) where you'll find help how to get these info most quickly (via the About page ⌘+I).

Also, assuming you've used the Add Translation window to add the key, if that didn't update the enum file, does also building the app not update the enum file (as you have the build script configured, it should also do the update)? If both don't work, you probably have a path issue, ensure youre config file path in the build script is correct and the path of the enum file within the config file is also correct.

**Reply by TopScrech** [reply]
Everything is now in order after regenerating the enum file and moving it to the Project Navigator. I appreciate your support!

**Reply by Jeehut** [reply]
Ah, that was the issue, should have guessed it 🤦‍♂️ but I'm glad you figured it out yourself! 👍
